projects/plugins/jetpack/class.jetpack.php

```diff
@@ -2033,15 +2033,21 @@ public static function load_modules() {
 			$modules = array_diff( $modules, $updated_modules );
 		}
 
+		$is_userless = self::connection()->is_userless();
+
 		foreach ( $modules as $index => $module ) {
-			// If we're in offline mode, disable modules requiring a connection.
-			if ( $is_offline_mode ) {
+			// If we're in offline/user-less mode, disable modules requiring a connection/user connection.
+			if ( $is_offline_mode || $is_userless ) {
 				// Prime the pump if we need to
 				if ( empty( $modules_data[ $module ] ) ) {
 					$modules_data[ $module ] = self::get_module( $module );
 				}
 				// If the module requires a connection, but we're in local mode, don't include it.
-				if ( $modules_data[ $module ]['requires_connection'] ) {
+				if ( $is_offline_mode && $modules_data[ $module ]['requires_connection'] ) {
+					continue;
+				}
+
+				if ( $is_userless && $modules_data[ $module ]['requires_user_connection'] ) {
```

Comment:
I'm not sure I understand this. Do we need those `$is_offline_mode` and `$is_userless` checks, given that we're inside the `if ( $is_offline_mode || $is_userless ) {` check above?

Reply:
Yes. The first `if ( $is_offline_mode || $is_userless ) ` check is just to share the `$modules_data[ $module ] = self::get_module( $module );` bit. But later we need to check for the specific conditions.

For example, If `$is_userless` is `false` and `requires_connection` is `true` we don't want to trigger the first `continue`.